### PR TITLE
Enable the useful readability-* checks in clang-tidy

### PR DESCRIPTION
**Pull request by steven-johnson**

### CODE REVIEW

File: src/CodeGen_LLVM.cpp [hunk 1]
```diff
@@ -3172,7 +3172,7 @@ void CodeGen_LLVM::visit(const Call *op) {
         }
 
         // Create a null-initialized global to track this object.
-        const auto base_fn = sub_fns.back().fn;
```
Comment (abadams) [inline src/CodeGen_LLVM.cpp:3175]: I suspect the original was a typo and the intent was for it to be `const auto *`

File: src/Dimension.h [hunk 2]
```diff
@@ -83,7 +83,7 @@ class Dimension {
     /** Get a different dimension of the same buffer */
     // @{
     Dimension dim(int i);
-    const Dimension dim(int i) const;
+    Dimension dim(int i) const;
```
Comment (abadams) [inline src/Dimension.h:86]: Wait... why do both these accessors exist if the const one is just as good as the non-const one?

**Review by abadams — APPROVED** [review]
lgtm with nits

File: src/Dimension.h (hunk 2 above)
Comment (steven-johnson) [reply src/Dimension.h:86]: They probably don't. I'll just delete the non-const one. The claim is: "readability-const-return-type:
Checks for functions with a const-qualified return type and recommends removal of the const keyword. Such use of const is usually superfluous, and can prevent valuable compiler optimizations."

**Pull request merged**